lib/shape.ml

```diff
@@ -2,9 +2,9 @@ open Graphics
 
 type point = { x : int; y : int }
 type line = { a : point; b : point }
-type rectangle = { c : point; length : int; width : int }
-type circle = { c : point; radius : int }
-type ellipse = {c : point; rx : int; ry: int}
+type circle = { c : point; radius : int; fill : (int * int * int) option; stroke : (int * int * int) option }
+type rectangle = { c : point; length : int; width : int; fill : (int * int * int) option; stroke : (int * int * int) option }
+type ellipse = { c : point; rx : int; ry: int; fill : (int * int * int) option; stroke : (int * int * int) option }
```

Comment:
1. Can you please make `fill` and `stroke` non optional?
We can explicitly default to `transparent` for `fill` and `black` for `stroke`.

2. Can you please create a new type for `color`? For now it can have two type constructors: `RGB of int * int * int` and `Transparent`.

Reply:
Can we make the type `color` something like `Transparent | Fill of int * int * int | Stroke of int * int * int`? That way we store this information in the same field in a shape. If both are active, Fill can override Stroke, I suppose.

Reply:
Okay, sure! Thank you

Reply:
> If both are active, Fill can override Stroke, I suppose.

@sudha247 I don't think this will let us create shapes with both a border and a fill.

Reply:
Yeah, fair enough. I should have considered the case where stroke and fill can be two colors. @nangahamandine, we should switch to the previous scheme. Sorry for the noise.

Reply:
> Yeah, fair enough. I should have considered the case where stroke and fill can be two colors. @nangahamandine, we should switch to the previous scheme. Sorry for the noise.

Oh sure! I'll redo my contribution. And thank you for the suggestion, it made me see the code from another perspective and challenged me to learn some new concepts.